### PR TITLE
chore(#8237): upgrade webapp's Angular to v16

### PR DESCRIPTION
**Pull request by latin-panda**
# Description

- Upgrades webapp's angular to v16
- Upgrades Typescript, rxjs, tslibs, ngxbootstrap versions in order to support Angular 16.
- Replaces Bootstrap's accordion with Angular Material's. This fixes this [issue](https://github.com/medic/cht-core/issues/8471).
  - Bootstrap's dropdown is the only component left, and it can be removed once we delete the support for the old design of actionbar and filters. The idea is to remove Bootstrap eventually.
- Removes `defaultProject` from angular.json, this property is deprecated.
- Because the new version of ngx-bootstrap drop support to Bootstrap v3, we started to use Bootstrap v4.
- Overrides styles of Bootstrap 4, patching, so app looks the same as before in the new design and old design (action bar and filters). We are working on removing Bootstrap gradually from Webapp. 
- Overrides Angular Material's fonts to use the Noto fonts, this fixes [this issue ](https://github.com/medic/cht-core/issues/8520)
- I tested in Chrome 116 and Chrome 90, seems to work fine. 

#### Performance   
**Build time**
- Angular 15:
  - `npm run build-dev` Best: 3m 01s — Average: 3m 07s — Worst: 4m 02s
- Angular 16:
  - `npm run build-dev` Best: 1m 19s —  Average: 2m 5s — Worst: 2m 37s
  
**Size**
- Angular 15:

Initial Chunk Files       |  Raw Size
--------|--------
main.js                     |  11.93 MB | 
polyfills.js          |   1.04 MB | 
styles.css            | 572.33 kB | 
scripts.js                | 152.52 kB | 
runtime.js          |   6.99 kB | 
  | **Initial Total** |  **13.68 MB**

- Angular 16:

Initial Chunk Files        |  Raw Size
--------|--------
main.js            |  12.30 MB | 
styles.css       | 537.11 kB | 
polyfills.js       | 230.55 kB | 
scripts.js          | 152.52 kB | 
runtime.js        |   6.99 kB | 
 | **Initial Total** |  **13.20 MB**

**Unit tests time**
- Angular 15:
  - `npm run unit-webapp` total: 2m 23s
- Angular 16:
  - `npm run unit-webapp` total: 1m 40s

https://github.com/medic/cht-core/issues/8237

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:angular_upgrade_to_v16_44/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:angular_upgrade_to_v16_44/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:angular_upgrade_to_v16_44/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: webapp/src/css/panel.less
Comment (latin-panda) [inline webapp/src/css/panel.less:1]: Unused styles as far I can see.

File: webapp/src/ts/app.component.ts [hunk 1]
```diff
@@ -142,7 +142,7 @@ export class AppComponent implements OnInit, AfterViewInit {
 
     this.adminUrl = this.locationService.adminPath;
 
-    setBootstrapTheme('bs3');
+    setBootstrapTheme('bs4');
```
Comment (latin-panda) [inline webapp/src/ts/app.component.ts:145]: ngx-bootstrap dropped support to bs3

**Comment by latin-panda** [comment]
Hi @jkuester and @ngaruko Can you please review? 
In the description, you can find more details. I tested with Chrome 90 and seems okay so far. 
Thanks for your time :)

**Comment by latin-panda** [comment]
@ngaruko do you have time to give it a run in one of your old Android phones? Bonus points if it has a Chrome version 90 or 95

**Comment by ngaruko** [comment]
Testing on local > **(WIP)**
Will test a few scenarios with the browser and android 13, 12, and 10.
Will attempt to find an older device (maybe use of the cloud providers for a few hours) with an old chrome version.

**Comment by jkuester** [comment]
This is random but it looks like for some reason the current user is showing up in the message drop-down when it was not before.  In both cases I have the same user/contact hierarchy and am logged in as `My First CHW`.

master:
image

angular_upgrade_to_v16_44:

image

**Review by jkuester — APPROVED** [review]
:tada: Awesome!  I spend a bunch of time walking through side-by-side comparisons of `master` vs this branch and everything looks great!  (The only difference I could find was that strange difference in the Messages tab, but maybe that is related to something weird in my local setup....)  

I even tried enabling both the `view_old...` flags and validated that everything still looks identical with the old functionality.  Great job!

**Comment by ngaruko** [comment]
Tested a few functionalities with chrome, android 13, 12, and 10 - and it all looks great and same as master.

<details><summary>Message tab</summary>
image

</details>

<details><summary>Contact tab > </summary>
creating/editing contacts

</details>

<details><summary>report tabs > </summary>
 creating editing reports 
image

</details>

<details><summary>Tasks tabs > </summary>
 tasks are created and can be completed
image

</details>

<details><summary>Targets tab </summary>
image

</details>
<details><summary>Warnings when existing form before completion </summary>
image

</details>

**Review by ngaruko — APPROVED** [review]
Looks good!

**Comment by latin-panda** [comment]
@jkuester I tried the message, and in both master and this, I get my logged-in user's contact appearing in the list as in your screenshot. I think it's okay

**Pull request merged**